helios-authentication/src/main/java/com/spotify/helios/auth/AuthenticationPlugin.java

```diff
@@ -45,8 +45,6 @@
 
   ServerAuthentication<C> serverAuthentication();
 
-  ClientAuthentication<C> clientAuthentication();
```

Comment:
The `AuthenticationPlugin` interface doesn't really make sense anymore, with `ClientAuthentication` removed. Remove one layer here for the server-side as well? E.g. `AuthenticationPlugin` -> `ServerAuthPlugin`